config/src/config/upstream_config.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_types::PeerId;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct UpstreamConfig {
+    // The upstream networks that this node belongs to as statically defined in the NodeConfig of this UpstreamConfig
+    // A network is considered upstream if all peers in this network are upstream w.r.t. this node
+    pub primary_networks: HashSet<PeerId>,
```

Comment:
I don't think we'll have such a scenario in the current setup, but I made it a HashSet just in case our assumptions about the possible network topologies change